site/app/views/grading/ElectronicGraderView.php

```diff
@@ -1138,11 +1143,28 @@ public function renderGradingPanelHeader(bool $isPeerPanel, bool $isStudentInfoP
      * @param bool $show_hidden_cases
      * @return string
      */
-    public function renderAutogradingPanel($version_instance, bool $show_hidden_cases) {
+    public function renderAutogradingPanel($version_instance, bool $show_hidden_cases, $graded_gradeable, $gradeable, $display_version) {
         $this->core->getOutput()->addInternalJs('submission-page.js');
+        $this->core->getOutput()->addInternalJs('drag-and-drop.js');
+        $this->core->getOutput()->addVendorJs('bootstrap/js/bootstrap.bundle.min.js');
+
+        //get user id for regrading, if team assignment user id is the id of the first team member, team id and who id will be determined later
+        if ($gradeable->isTeamAssignment()) {
+            $id = $graded_gradeable->getSubmitter()->getTeam()->getMemberUsers()[0]->getId();
+        }
+        else {
+            $id = $graded_gradeable->getSubmitter()->getId();
+        }
+
         return $this->core->getOutput()->renderTwigTemplate("grading/electronic/AutogradingPanel.twig", [
             "version_instance" => $version_instance,
             "show_hidden_cases" => $show_hidden_cases,
+            "highest_version" =>  $graded_gradeable->getAutoGradedGradeable()->getHighestVersion(),
+            "max_submissions" => $gradeable->getAutogradingConfig()->getMaxSubmissions(),
+            "is_vcs" => $gradeable->isVcs(),
+            "gradeable_id" => $gradeable->getId(),
```

Comment:
Do you need to pass in `$gradeable` here, or can we just do `$graded_gradeable->getGradeable()`?

Reply:
Nope you're right